packages/cli/src/commands/init/init.js

```diff
@@ -127,23 +175,22 @@ async function installDependencies({
   loader.succeed();
 }
 
-function createProject(projectName: string, options: Options, version: string) {
-  fs.mkdirSync(projectName);
-  process.chdir(projectName);
-
-  if (options.template) {
-    return createFromTemplate({
-      projectName,
-      templateName: options.template,
-      npm: options.npm,
-    });
-  }
+async function createProject(
+  projectName: string,
+  directory: string,
+  version: string,
+  options: Options,
+) {
+  const templateName = options.template || 'react-native';
 
   return createFromTemplate({
     projectName,
-    templateName: 'react-native',
-    version,
+    templateName,
+    // version is "latest" by default, but it's easier for us to treat it as
+    // undefined when the "template" param is passed. Might refactor later
+    version: options.template ? undefined : version,
```

Comment:
I don't like this change. I think an `if` statement (calling `createFromTemplate` when `template` is passed) was clearer and more elegant.

What was the reason for this change?

Reply:
less repetition mainly

Reply:
I find it confusing, specially that we leave "might refactor later". I'd suggest "let's change it later" or make it clear right now :)